### DISCUSSION TITLE
Cache results from API calls between pages with SSG/getStaticProps+revalidate

### DISCUSSION THREAD

**Discussion by tobbbe** [vercel/next.js#20200] [Help]
We have a site where we list links to ~100 subpages in the header menu. They are fetched from our server (`/api/subpages`) in getStaticProps in _app.js.

Every time revalidate re-generation runs our backend gets one request to `/api/subpages` for each subpage, that's 100*100=10000 requests!

I've tried to cache the api calls in memory but each page re-generation has it's own memory (most of the time it seems).

Is there anyway to cache the api-calls or only make them once and then share between re-generations?

**Comment by elibenton** [comment]
I am having a very similar issue, but I am calling an external API in getStaticProps. I quickly reach the API's limit per minute, and the build fails.

I would happily cache the values if there were a way.
